sympy/solvers/ode/riccati.py

```diff
@@ -0,0 +1,791 @@
+
+    ..math :: 1 - x = (x - 4) \Sum_{x = 0}^{\infty} a[i] x^i
+
+    Multiplying this out and rearranging the indices of the
+    sums, we get -
+
+    ..math :: 1 - x = -4 a_0 + (a_0 - 4 a_1) x + \Sum_{x = 2}^{\infty} (a[i-1] - 4*a[i]) x^i
+
+    Find the base case coefficients and then the recurrence
+    relation. Solving for these, we get -
+
+    ..math:: a_0 = -\frac{1}{4}, a_1 = {3}{16}, a_m = \frac{a_{m-1}}{4}
+    """
+    one = Poly(1, x, extension=True)
+
+    if r == oo:
+        # Series at x = oo is equal to first transforming
+        # the function from x -> 1/x and finding the
+        # series at x = 0
+        num, den = inverse_transform_poly(num, den, x)
+        r = S(0)
+
+    if r:
+        # For an expansion about a non-zero point, a
+        # transformation from x -> x + r must be made
+        num = num.transform(Poly(x + r, x, extension=True), one)
+        den = den.transform(Poly(x + r, x, extension=True), one)
+        num, den = num.cancel(den, include=True)
+
+    # Remove the pole from the denominator if the series
+    # expansion is about one of the poles
+    num, den = (num*x**m).cancel(den, include=True)
+
+    # Equate coefficients for the first terms (base case)
+    maxdegree = 1 + max(num.degree(), den.degree())
+    syms = symbols(f'a:{maxdegree}', cls=Dummy)
+    diff = num - den * Poly(syms[::-1], x)
+    coeff_diffs = diff.all_coeffs()[::-1][:maxdegree]
+    (coeffs, ) = linsolve(coeff_diffs, syms)
+
+    # Use the recursion relation for the rest
+    recursion = den.all_coeffs()[::-1]
+    div, rec_rhs = recursion[0], recursion[1:]
+    series = list(coeffs)
+    while len(series) < n:
+        next_coeff = sum(c*series[-1-n] for n, c in enumerate(rec_rhs)) / div
+        series.append(-next_coeff)
+    series = {m - i: val for i, val in enumerate(series)}
+    return series
+
+def compute_m_ybar(x, poles, choice, c, d, N):
+    ybar = 0
+    m = Poly(d[choice[0]][-1], x, extension=True)
+
+    # Calculate the first (nested) summation for ybar
+    # as given in Step 9 of the Thesis (Pg 82)
+    for i in range(len(poles)):
+        for j in range(len(c[i][choice[i + 1]])):
+            ybar += c[i][choice[i + 1]][j]/(x - poles[i])**(j+1)
+        m -= Poly(c[i][choice[i + 1]][0], x, extension=True)
+
+    # Calculate the second summation for ybar
+    for i in range(N+1):
+        ybar += d[choice[0]][i]*x**i
+    return (m.expr, ybar)
+
+
+def solve_aux_eq(numa, dena, numy, deny, x, m):
+    # Assume that the solution is of the type
+    # p(x) = C0 + C1*x + ... + Cm*x**m
+    psyms = symbols(f'C0:{m}', cls=Dummy)
+    domain = ZZ[psyms]
+    psyms += (1, )
+    psol = 0
+    for i in range(m + 1):
+        psol += Poly(psyms[i]*x**i, x, domain=domain)
+
+    # Eq (5.16) in Thesis - Pg 81
+    auxeq = (dena*(numy.diff(x)*deny - numy*deny.diff(x) + numy**2) - numa*deny**2)*psol
+    if m >= 1:
+        px = psol.diff(x)
+        auxeq += 2*px*numy*deny*dena
```

Comment:
If `px` is a much larger polynomial than the others then it will be more efficient to compute this as
```python
auxeq += px*(2*numy*deny*dena)
```
In other words do most multiplication with small polynomials and only one multiplication with the large polynomial.

Reply:
Depending on the value of `m`, `px` may be large enough. I will change it.